### PR TITLE
fix: TimeoutQC aggregation

### PR DESCRIPTION
**Pull request by pompon0**
Custom Ord implementation was causing timeout votes with different high qc (with the same view) to be deduplicated, which invalidated the aggregated signature. I've added a test for this case, and preemptively removed all custom Ord implementations on messages.

### CODE REVIEW

File: node/components/bft/src/chonky_bft/mod.rs [hunk 1]
```diff
@@ -276,7 +275,14 @@ impl StateMachine {
         ctx: &ctx::Ctx,
         qc: &validator::CommitQC,
     ) -> ctx::Result<()> {
-        self.high_commit_qc = max(Some(qc.clone()), self.high_commit_qc.clone());
+        if self
+            .high_commit_qc
+            .as_ref()
+            .map_or(false, |old| old.view().number >= qc.view().number)
```
Comment (brunoffranca) [inline node/components/bft/src/chonky_bft/mod.rs:281]:
```suggestion
            .map_or(false, |old| old.view().number > qc.view().number)
```
Nit: Greater than should be enough. If we have two QCs at the same view, they must be for the same block.

File: node/components/bft/src/chonky_bft/proposal.rs [hunk 2]
```diff
@@ -223,6 +221,7 @@ impl StateMachine {
 
         // Update the state machine.
         self.view_number = message.view().number;
+        metrics::METRICS.replica_view_number.set(self.view_number.0);
```
Comment (brunoffranca) [inline node/components/bft/src/chonky_bft/proposal.rs:224]: Why move the start of the view to here?

**Review by brunoffranca — DISMISSED** [review]

File: node/components/bft/src/chonky_bft/proposal.rs (hunk 2 above)
Comment (pompon0) [reply node/components/bft/src/chonky_bft/proposal.rs:224]: I'm moving the metrics updates next to the places where the variable that they track is updated. In particular, before this pr one of such places has been missed.

File: node/components/bft/src/chonky_bft/mod.rs (hunk 1 above)
Comment (pompon0) [reply node/components/bft/src/chonky_bft/mod.rs:281]: if old view == qc view, there is nothing to be done, so IMO this WAI

**Pull request merged**